### PR TITLE
python script for extraction

### PR DESCRIPTION
**Pull request by sujanparab12**
First push of python script for extraction

### CODE REVIEW

**Review by OmkarShevde — COMMENTED** [review]
Please keep only one extraction script to avoid confusion.

**Review by OmkarShevde — APPROVED** [review]

File: scripts/python/config/keydata/config.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# config.yaml
+
+source:
+  username: '<user_name>'
+  password: '<password>'
+  server: '<server_name>'
+  database: '<database_name>'
```
Comment (omingale0707) [inline scripts/python/config/keydata/config.yaml:7]: Move this to key.yml

File: scripts/python/config/keydata/key.yaml
Comment (omingale0707) [inline scripts/python/config/keydata/key.yaml:1]: change the file name to credentials.yml

File: scripts/python/python_extract_load_to_s3.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    # S3 Bucket details and local directory
+    bucket_name = 'eltproject'
+    local_root_directory = 'C:/Users/durge/GIT project/de-batch-project/new_output_batch'
+
+    # Load credentials from the YAML file
+    credentials = S3Uploader().load_credentials(aws_credentials_path)
+    
+    # Extract AWS credentials from the loaded credentials
+    aws_access_key_id = credentials['keys']['access_key_id']
+    aws_secret_access_key = credentials['keys']['secret_access_key']
```
Comment (omingale0707) [inline scripts/python/python_extract_load_to_s3.py:99]: As discussed yesterday, I don't see the changes made to move this in class itself. Make sure the attributes in class are hidden

File: scripts/python/python_extract_load_to_s3.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            s3_client.upload_file(local_file_path, s3_bucket_name, s3_key)
+            print(f'Uploaded {filename} to {s3_bucket_name}/{s3_key}')
+                
+def main():
+    # Paths to configuration and information files
+    config_path = 'C:\\Users\\durge\\GIT project\\de-batch-project\\scripts\\python\\config\\keydata\\config.yaml'
+    aws_credentials_path = 'C:\\Users\\durge\\GIT project\\de-batch-project\\scripts\\python\\config\\keydata\\key.yaml'
+    
+    # S3 Bucket details and local directory
+    bucket_name = 'eltproject'
```
Comment (omingale0707) [inline scripts/python/python_extract_load_to_s3.py:91]: Move this to config file

File: scripts/python/python_extract_load_to_s3.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            start_index = i * batch_size
+            end_index = (i + 1) * batch_size
+            batch_data = df.iloc[start_index:end_index]
+            output_file_path = f"{output_directory}/{table}_{i + 1:02d}.csv"
+            batch_data.to_csv(os.path.join(output_file_path), index=False)
+        
+            
+
+    def fetch_data(self, table):
+        sql_query = f'SELECT top 2000* FROM {table}'
```
Comment (omingale0707) [inline scripts/python/python_extract_load_to_s3.py:43]: Why is there top 2000? The code should not contain these limitations

**Pull request closed without merging**